src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -1,205 +1,117 @@
-                "azure-arm":
-                    sdkContext.arm === false ? undefined : sdkContext.arm,
-                "disable-xml-docs":
-                    options["disable-xml-docs"] === false
-                        ? undefined
-                        : options["disable-xml-docs"]
-            };
-
-            await program.host.writeFile(
-                resolvePath(generatedFolder, configurationFileName),
-                prettierOutput(JSON.stringify(configurations, null, 2))
-            );
-
-            if (options.skipSDKGeneration !== true) {
-                const csProjFile = resolvePath(
-                    outputFolder.endsWith("src")
-                        ? outputFolder
-                        : resolvePath(outputFolder, "src"),
-                    `${configurations["library-name"]}.csproj`
-                );
-                Logger.getInstance().info(`Checking if ${csProjFile} exists`);
-                const newProjectOption =
-                    options["new-project"] || !existsSync(csProjFile)
-                        ? "--new-project"
-                        : "";
-                const existingProjectOption = options["existing-project-folder"]
-                    ? `--existing-project-folder ${options["existing-project-folder"]}`
-                    : "";
-                const debugFlag = options.debug ?? false ? " --debug" : "";
-
-                const command = `dotnet --roll-forward Major ${resolvePath(
-                    options.csharpGeneratorPath
-                )} --project-path ${outputFolder} ${newProjectOption} ${existingProjectOption} --clear-output-folder ${
-                    options["clear-output-folder"]
-                }${debugFlag}`;
-                Logger.getInstance().info(command);
-
-                try {
-                    execSync(command, { stdio: "inherit" });
-                } catch (error: any) {
-                    if (error.message) Logger.getInstance().info(error.message);
-                    if (error.stderr) Logger.getInstance().error(error.stderr);
-                    if (error.stdout)
-                        Logger.getInstance().verbose(error.stdout);
-                    throw error;
-                }
-            }
-
-            if (!options["save-inputs"]) {
-                // delete
-                deleteFile(resolvePath(generatedFolder, tspOutputFileName));
-                deleteFile(resolvePath(generatedFolder, configurationFileName));
-            }
+        configurations["shared-source-folders"] = resolvedSharedFolders ?? [];
+        configurations["flavor"] =
+            options["flavor"] ??
+            (configurations.namespace.toLowerCase().startsWith("azure.")
+                ? "azure"
+                : undefined);
+        /* set azure-arm */
+        const sdkContext = createSdkContext(
+            context,
+            "@azure-tools/typespec-csharp"
+        );
+        configurations["azure-arm"] =
```

Comment:
I heard that there was a TODO item about ARM to parse it from the decorators list?
If so, could you add a TODO here?

Reply:
Yes. Long-term goal we will parse this from decorator list or maybe other approach. TODO added